lib/modern_searchlogic/column_conditions.rb

```diff
@@ -84,8 +92,17 @@ def searchlogic_suffix_condition_match(method_name)
             arity: arity,
             block: lambda do |*args|
               validate_argument_count!(arity, args.length) if arity >= 0
-              arel_conditions = column_names.map { |n| instance_exec(n, *args, &method_block) }.reduce(:or)
-              where(arel_conditions)
+              partial_queries = column_names.map do |n|
+                query = unscoped { instance_exec(n, *args, &method_block) }
+                if ActiveRecord::VERSION::MAJOR >= 5 && query.respond_to?(:where_clause)
+                  query.where_clause&.ast
+                elsif [3, 4].include?(ActiveRecord::VERSION::MAJOR) && query.respond_to?(:where_values)
+                  query.where_values
```

Comment:
for polymorphic associations, would this be something like `(event_target_id = 123019 AND event_target_type = 'Song')`?

Reply:
nevermind, we don't support associations in searchlogic.